src/RemixDevTools/RemixDevTools.tsx

```diff
@@ -23,9 +23,15 @@ interface Props {
 }
 
 const RemixDevTools = ({ defaultOpen, position }: Props) => {
-  const [isOpen, setIsOpen] = useState(defaultOpen);
-  const { activeTab, setActiveTab, setShouldConnectWithForge, height } =
-    useRDTContext();
+  const {
+    openState,
+    setOpenState,
+    activeTab,
+    setActiveTab,
+    setShouldConnectWithForge,
+    height,
+  } = useRDTContext();
+  const isOpen = defaultOpen || openState;
```

Comment:
because this is moved out of state wouldn't this mean even if the defaultOpen is false it would still be open i the openState is set to true? also this should be moved to session storage instead of local one, we don't want this setting persisted forever but only in the current session. Also I am surprised you've been getting these issues, usually the dev tools are rendered only once in the root

Reply:
It's similar of how react-query devtools works.
If you want it per session, I can work on it.

Reply:
sorry I've been a bit busy, I will take a deeper dive into this very soon

Reply:
we will probably move this into session

Reply:
@AhmedNawaf mind recording a demo video of the issue, just to see the exact case, I am assuming when you say page reload you mean a full reload triggered with something like window.refresh or something?